python/lsst/cp/pipe/ptc.py

```diff
@@ -706,122 +420,338 @@ def measureMeanVarCov(self, exposure1, exposure2, region=None, covAstierRealSpac
                           f"({self.config.minNumberGoodPixelsForFft})")
             return np.nan, np.nan, None
 
-        maxRangeCov = self.config.maximumRangeCovariancesAstier
-        if covAstierRealSpace:
-            covDiffAstier = computeCovDirect(diffIm.getImage().getArray(), w, maxRangeCov)
-        else:
-            shapeDiff = diffIm.getImage().getArray().shape
-            fftShape = (fftSize(shapeDiff[0] + maxRangeCov), fftSize(shapeDiff[1]+maxRangeCov))
-            c = CovFft(diffIm.getImage().getArray(), w, fftShape, maxRangeCov)
-            covDiffAstier = c.reportCovFft(maxRangeCov)
+        maxRangeCov = self.config.maximumRangeCovariancesAstier
+        if covAstierRealSpace:
+            covDiffAstier = computeCovDirect(diffIm.getImage().getArray(), w, maxRangeCov)
+        else:
+            shapeDiff = diffIm.getImage().getArray().shape
+            fftShape = (fftSize(shapeDiff[0] + maxRangeCov), fftSize(shapeDiff[1]+maxRangeCov))
+            c = CovFft(diffIm.getImage().getArray(), w, fftShape, maxRangeCov)
+            covDiffAstier = c.reportCovFft(maxRangeCov)
+
+        return mu, varDiff, covDiffAstier
+
+
+class PhotonTransferCurveSolveConnections(pipeBase.PipelineTaskConnections,
+                                          dimensions=("instrument", "detector")):
+    inputCovariances = cT.Input(
+        name="ptcCovariances",
+        doc="Tuple with measured covariances from flats.",
+        storageClass="PhotonTransferCurveDataset",
+        dimensions=("instrument", "exposure", "detector"),
+        multiple=True,
+    )
+    camera = cT.PrerequisiteInput(
+        name="camera",
+        doc="Camera the input data comes from.",
+        storageClass="Camera",
+        dimensions=("instrument",),
+        isCalibration=True,
+        lookupFunction=lookupStaticCalibration,
+    )
+    outputPtcDataset = cT.Output(
+        name="ptcDatsetProposal",
+        doc="Output proposed ptc dataset.",
+        storageClass="PhotonTransferCurveDataset",
+        dimensions=("instrument", "detector"),
+        multiple=False,
+        isCalibration=True,
+    )
+
+
+class PhotonTransferCurveSolveConfig(pipeBase.PipelineTaskConfig,
+                                     pipelineConnections=PhotonTransferCurveSolveConnections):
+    """Configuration for fitting measured covariances.
+    """
+    ptcFitType = pexConfig.ChoiceField(
+        dtype=str,
+        doc="Fit PTC to Eq. 16, Eq. 20 in Astier+19, or to a polynomial.",
+        default="POLYNOMIAL",
```

Comment:
If this is POLYNOMIAL here, and not overridden in the pipelines/measurePhotonTransferCurve.yaml, then this will lead to inconsistencies between the ci_cpp_gen2 and ci_cpp_gen3 versions.  I'll make sure there are notes on both of those PRs as well.

Reply:
I'll change the value to {{FULLCOVARIANCE}} in ci_ccp_gen2 and in {{cp_pipe/pipelines/measurePhotonTransferCurve.yaml}}  (which will be used by ci_cpp_gen3)